<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>1 Introduction to IMAP Framework | IMAP: Integrated Microbiome Analysis Pipelines</title>
<meta name="author" content="Teresia Mrema-Buza">
<meta name="description" content="Welcome to the IMAP (“Integrated Microbiome Analysis Pipelines”) framework—a meticulously crafted structure empowering researchers, scientists, developers, and enthusiasts in conducting thorough...">
<meta name="generator" content="bookdown 0.37 with bs4_book()">
<meta property="og:title" content="1 Introduction to IMAP Framework | IMAP: Integrated Microbiome Analysis Pipelines">
<meta property="og:type" content="book">
<meta property="og:url" content="https://tmbuza.github.io/imap-project-overview/imap-framework.html">
<meta property="og:image" content="https://tmbuza.github.io/imap-project-overview/images/planning.png">
<meta property="og:description" content="Welcome to the IMAP (“Integrated Microbiome Analysis Pipelines”) framework—a meticulously crafted structure empowering researchers, scientists, developers, and enthusiasts in conducting thorough...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="1 Introduction to IMAP Framework | IMAP: Integrated Microbiome Analysis Pipelines">
<meta name="twitter:description" content="Welcome to the IMAP (“Integrated Microbiome Analysis Pipelines”) framework—a meticulously crafted structure empowering researchers, scientists, developers, and enthusiasts in conducting thorough...">
<meta name="twitter:image" content="https://tmbuza.github.io/imap-project-overview/images/planning.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Sen-0.4.8/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/Bitter-0.4.8/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.6.1/transition.js"></script><script src="libs/bs3compat-0.6.1/tabs.js"></script><script src="libs/bs3compat-0.6.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 1em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides Using Integrated Approaches"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a>:
        <small class="text-muted">End-to-End Practical User Guides Using Integrated Approaches</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Microbiome Explorer’s Practical User Guides</a></li>
<li><a class="" href="license.html">License</a></li>
<li class="book-part">IMAP PROJECT</li>
<li><a class="" href="about-imap-project.html">About IMAP Project</a></li>
<li><a class="" href="mission-and-vision.html">Mission and Vision</a></li>
<li><a class="" href="acknowledgements.html">Acknowledgements</a></li>
<li><a class="" href="dedication.html">Dedication</a></li>
<li class="book-part">AUTHOR’S PERSPECTIVE</li>
<li><a class="" href="motivation-and-commitment.html">Motivation and Commitment</a></li>
<li class="book-part">IMAP FRAMEWORK</li>
<li><a class="active" href="imap-framework.html"><span class="header-section-number">1</span> Introduction to IMAP Framework</a></li>
<li class="book-part">IMAP-PART 01</li>
<li><a class="" href="imap-part01.html"><span class="header-section-number">2</span> IMAP Toolkit: Essential Requirements</a></li>
<li class="book-part">IMAP-PART 02</li>
<li><a class="" href="imap-part02.html"><span class="header-section-number">3</span> Microbiome sample metadata</a></li>
<li class="book-part">IMAP-PART 03</li>
<li><a class="" href="imap-part03.html"><span class="header-section-number">4</span> Downloading Reads from SRA</a></li>
<li class="book-part">IMAP-PART 04</li>
<li><a class="" href="imap-part04.html"><span class="header-section-number">5</span> Quality Control of Fastq Reads</a></li>
<li class="book-part">IMAP-PART 05</li>
<li><a class="" href="imap-part05.html"><span class="header-section-number">6</span> Microbial profiling using Mothur</a></li>
<li class="book-part">IMAP-PART 06</li>
<li><a class="" href="imap-part06.html"><span class="header-section-number">7</span> Microbial profiling using QIIME2</a></li>
<li class="book-part">IMAP-PART 07</li>
<li><a class="" href="imap-part07.html"><span class="header-section-number">8</span> Data preprocessing</a></li>
<li class="book-part">IMAP-PART 08</li>
<li><a class="" href="imap-part08.html"><span class="header-section-number">9</span> Data Exploration</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></li>
<li><a class="" href="session-information.html"><span class="header-section-number">B</span> Session Information</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/imap-project-overview.git">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="imap-framework" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> Introduction to IMAP Framework<a class="anchor" aria-label="anchor" href="#imap-framework"><i class="fas fa-link"></i></a>
</h1>
<p>Welcome to the IMAP (“Integrated Microbiome Analysis Pipelines”) framework—a meticulously crafted structure empowering researchers, scientists, developers, and enthusiasts in conducting thorough analyses of intricate microbiome data. Each chapter within IMAP stands as an independent GitHub repository (repo), offering a dedicated space for focused exploration and understanding.</p>
<p>GitHub Repo: URL: <a href="https://github.com/tmbuza/imap-project-overview/" class="uri">https://github.com/tmbuza/imap-project-overview/</a> <br></p>
<p>GitHub Pages: URL: <a href="https://tmbuza.github.io/imap-project-overview/" class="uri">https://tmbuza.github.io/imap-project-overview/</a></p>
<div id="primary-objectives" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Primary objectives<a class="anchor" aria-label="anchor" href="#primary-objectives"><i class="fas fa-link"></i></a>
</h2>
<p>Our primary objectives revolve around reproducibility, clarity, and efficiency. We understand the challenges inherent in microbiome data analysis and aim to equip you with the tools and strategies necessary to navigate this complex landscape with confidence.</p>
</div>
<div id="environments-for-streamlined-analysis" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Environments for Streamlined Analysis<a class="anchor" aria-label="anchor" href="#environments-for-streamlined-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>Microbiome data analysis encompasses a diverse set of tools and platforms, from R and Python for statistical analysis to Snakemake for workflow management, and GitHub Actions for continuous integration and deployment. To streamline this process and ensure reproducibility, we adopt a strategy of managing these tools within unified environments.</p>
<ul>
<li><p><strong>RStudio for R Analysis:</strong> Discover how we leverage RStudio for developing and documenting R scripts or notebooks. All required R packages are installed and managed within the R environment, providing a controlled and reproducible setup.</p></li>
<li><p><strong>Jupyter Notebook for Python Analysis:</strong> Dive into the use of Jupyter Notebooks for Python-based analysis. Python packages are isolated within the Jupyter environment, ensuring a clean and consistent setup.</p></li>
<li><p><strong>Conda Environment for Snakemake Workflows:</strong> Explore the versatility of Snakemake workflows, managed within a dedicated Conda environment. This approach guarantees that dependencies for each workflow are isolated and reproducible.</p></li>
<li><p><strong>GitHub Actions for Continuous Integration and Deployment:</strong> Learn how we utilize GitHub Actions to automate continuous integration and deployment processes. This ensures the consistent and automated testing, building, and deployment of our microbiome data analysis workflows, contributing to a more efficient and reliable development pipeline.</p></li>
</ul>
</div>
<div id="analysis-outputs" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> Analysis Outputs<a class="anchor" aria-label="anchor" href="#analysis-outputs"><i class="fas fa-link"></i></a>
</h2>
<p>Outputs from RStudio, Jupyter Notebook, and Snakemake workflows converge in a final HTML report, serving as the culmination of our analytical efforts.</p>
<ul>
<li><p><strong>Comprehensive Insights:</strong> The final HTML report offers a holistic view of the microbiome analysis, incorporating both the code and results from various components.</p></li>
<li><p><strong>Efficient Collaboration:</strong> Discover how consolidating outputs enhances collaboration, enabling team members to easily reproduce and build upon the analysis.</p></li>
</ul>
</div>
<div id="snakemake-rule-graph-a-tool-for-transparency" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> Snakemake Rule Graph: A Tool for Transparency<a class="anchor" aria-label="anchor" href="#snakemake-rule-graph-a-tool-for-transparency"><i class="fas fa-link"></i></a>
</h2>
<p>A pivotal aspect of our methodology involves the utilization of Snakemake rule graphs, serving as invaluable tools to enhance transparency in our computational analyses. These graphical representations provide a lucid and intuitive portrayal of the logical sequence of tasks, contributing to a heightened understanding for users, readers, collaborators, or reviewers.</p>
<p>We encourage readers to make use of the Snakemake rule graphs featured in the appendix section of chapters dedicated to intensive computational analyses. These visual aids stand as comprehensive resources, aiding in the navigation and comprehension of the intricate structure and dependencies embedded within each analysis.</p>
</div>
<div id="considerations-for-incompatibility" class="section level2" number="1.5">
<h2>
<span class="header-section-number">1.5</span> Considerations for Incompatibility<a class="anchor" aria-label="anchor" href="#considerations-for-incompatibility"><i class="fas fa-link"></i></a>
</h2>
<p>While advocating for a unified environment, we acknowledge that certain packages may be incompatible with others. We offer guidance on handling such scenarios, ensuring that potential conflicts are addressed with minimal disruption to the analysis process.</p>
</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="motivation-and-commitment.html">Motivation and Commitment</a></div>
<div class="next"><a href="imap-part01.html"><span class="header-section-number">2</span> IMAP Toolkit: Essential Requirements</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#imap-framework"><span class="header-section-number">1</span> Introduction to IMAP Framework</a></li>
<li><a class="nav-link" href="#primary-objectives"><span class="header-section-number">1.1</span> Primary objectives</a></li>
<li><a class="nav-link" href="#environments-for-streamlined-analysis"><span class="header-section-number">1.2</span> Environments for Streamlined Analysis</a></li>
<li><a class="nav-link" href="#analysis-outputs"><span class="header-section-number">1.3</span> Analysis Outputs</a></li>
<li><a class="nav-link" href="#snakemake-rule-graph-a-tool-for-transparency"><span class="header-section-number">1.4</span> Snakemake Rule Graph: A Tool for Transparency</a></li>
<li><a class="nav-link" href="#considerations-for-incompatibility"><span class="header-section-number">1.5</span> Considerations for Incompatibility</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/imap-project-overview.git/blob/gh-pages/01-imap-framework.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/imap-project-overview.git/edit/gh-pages/01-imap-framework.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">
      <div class="col-12 col-md-6 mt-3">
        <p><strong><u><a class="text-light" href="https://microbiome.complexdatainsights.com"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a></u></strong>: End-to-End Practical User Guides Using Integrated Approaches</p>
<br><p>Authored by Teresia Mrema-Buza, Last updated: <b>2024-02-17</b>.</p>
<br>
</div>
      <div class="col-12 col-md-6 mt-3">
        <p>Note: This practical guide was created using the <a class="text-light" href="https://bookdown.org/yihui/bookdown/">bookdown</a> R package, styled with the modern bs4_book format.</p>
      </div>

    </div>
  </div>
</footer>
</body>
</html>
